src/shared/components/Modal.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+const Modal = (props) => {
+    const findByKey = (name) =>
+        props.children.map((child) => {
+            if (child.key === name) return child;
+        });
+
+    const closeModal = (e) => {
+        e.stopPropagation();
+
+        if (e.target.classList.contains('modal-close')) {
+            return props.click();
+        }
+    };
+
+    return (
+        <div className="modal-mask modal-close" onClick={closeModal}>
+            <div className="modal-wrapper">
+                <div className="modal-container">
+                    <div className="modal-header">{findByKey('header')}</div>
+
+                    <div className="modal-body">{findByKey('body')}</div>
+
+                    <div className="modal-footer">
+                        <button className="modal-close" onClick={closeModal}>
+                            CLOSE
+                        </button>
+                        {findByKey('footer')}
+                    </div>
+                </div>
+            </div>
+        </div>
+    );
+};
+
+export default Modal;
```

Comment:
What type of Modal is this, Close something?

Reply:
It's a customize modal, so that this modal could do anything to send request to server (updating user, adding friend, showing details of sth, etc.). 
**Let take a look at FriendList component**. You just need to clarify key like **'header', 'body',** and **'footer'**. Then you could design the modal to do the thing you want it to do.